### PR TITLE
Login

### PR DESCRIPTION
**Pull request by haniscape**
Hello Josip , please review this draft PR , 
I am about to make last step of adding Sign In / New User tabs inside   login/login.html page

### CODE REVIEW

File: about-us/about-us.html [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+          <a href="https://www.facebook.com/bordersnone/"
+            ><i class="fab fa-facebook"></i
+          ></a>
+          <a href="https://www.instagram.com/bordersnone/"
+            ><i class="fab fa-instagram"></i
+          ></a>
+        </div>
+      </div>
+    </footer>
+    <script src="book.js"></script>
```
Comment (josipgrgic) [inline about-us/about-us.html]: Is a script reference to book.js needed in the about-us html file?

File: app.js [hunk 2]
```diff
@@ -16,13 +16,75 @@ function getBooks() {
...
+
+// Validating the entered data and show User Name on top
+
+document.getElementById('btn-submit').addEventListener('click', validate);
+
+function validate(e) {
+  e.preventDefault();
+  // ELEMENTS variables for username and password
+  let userName = document.getElementById('uname');
+  let passWord = document.getElementById('psw');
```
Comment (josipgrgic) [inline app.js:57]: `username `and `password`. Not `userName `and `passWord`

Comment (josipgrgic) [reply app.js:57]: and use full names for ids and not abbreviations. So not `psw` and `uname`

Comment (josipgrgic) [inline app.js]: userName is not correct. 
First, this variable is a reference to the html element and not the actual value of the username. 
Second, it's spelled `username `and not `userName`, but this should be called something like `usernameInput`

The same goes fot the password element

Comment (josipgrgic) [reply app.js]: and use full names for ids and not abbreviations. So not psw and uname

Comment (josipgrgic) [reply app.js]: The same comment goes for the other variables too. Change them accordingly, please

File: books/book.html [hunk 3]
```diff
@@ -12,19 +16,94 @@
...
+
+            <label for="uname">Username</label>
+            <input
+              type="text"
+              placeholder="Enter Username"
+              id="uname"
+              required
+            />
+
+            <label for="password">Password</label>
```
Comment (josipgrgic) [inline books/book.html]: Wrong id of the input element. As I said in a previous comment, I would like to see full names so keep this as is but change the id of the input element

**Comment by josipgrgic** [comment]
The about-us.css file is empty. Why is that?

**Comment by josipgrgic** [comment]
Why is there a login popup even though we agreed we are going to make only the simple login form inside login.html? And the login is only handled inside the landing page. 
I cannot login from the book details page.
When I login in the langing page, I see my login username in the navbar, but when I got to the book details page the info is suddenly dissapeared.

**Comment by josipgrgic** [comment]
I see that the PR is still in its draft state so I suppose it is still a work in progress. But, why are we still repeating the same issues over and over again? Why do I see a PR called Login contain changes not related to the login functionality? Why does this PR include login forms when we agreed on the call to keep it as simple as possible with only a login.html page and login.css and login.js?

**Comment by haniscape** [comment]
Dear Josip,
by the time we talked I had that popup already , so I thought it wouldnt hurt to include it beside what we have agreed upon (login.html)
As for other changes , they are mostly related to UI, which I started in first phase of the project, so I only cleared the repeated code and cleaned it .

File: app.js (hunk 2 above)
Comment (haniscape) [reply app.js]: alright got you and updating right now

**Comment by josipgrgic** [comment]
Hi Hany,

I understand, but the popup only complicates things. I really do not mind if it works correctly but I tested your branch and it is not. The approach we agreed upon was more simple with less code to make it work. So I give you free hands on this one even though I do not like changing stuff on your own. That is certainly not something that you will be doing when working for a client. If they have specific desires, our job is to either convince them there are better ways or accommodate them, but never do things on our own.

I know what the other changes are but I do not like them being in the same pull request as this login part. Please remove any code that is not related to the login functionality. I want you guys to learn that these types of things are not best practices and they only cause headaches to the reviewers of your code. The name of the branch is LOGIN I only want LOGIN RELATED CODE. You can make another PR called DESIGN or something more specific and have that code in it. You don't have to delete the code, just make a copy of the changed files somewhere outside the repository, remove the changes from this branch and add them back to the repo in the new branch

**Comment by haniscape** [comment]
Dear Josip, 
I finished migrating from pop up form into simple Login.html that can now show username across the pages of :
index, book, about-us and login 

I did some minor styling but all of them are related to the Login branch as per your instruction.

Please review my Pull Request

File: login/login.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    );
+    errorMsgDiv.append(errorMsgNode);
+    passwordInput.after(errorMsgDiv);
+    setTimeout(() => {
+      errorMsgDiv.innerText = '';
+    }, 1000);
+  } else {
+    // store the input into Session Storage
+    sessionStorage.setItem('sessionUsername', usernameValue);
+    sessionStorage.setItem('sessionPassword', passwordValue);
```
Comment (josipgrgic) [inline login/login.js]: The keys should be username and password instead of sessionUsername, sessionPassword

File: login/login.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    window.open('../login/login.html', '_self');
```
Comment (josipgrgic) [inline login/login.js]: After a successful login, the user should be redirected to the main page (index.html)

**Comment by josipgrgic** [comment]
Good job hany! Just fix the little thing I pointed out and then you can merge the PR into main

**Review by josipgrgic — APPROVED** [review]

File: login/login.js (hunk 4 above)
Comment (haniscape) [reply login/login.js]: I did it , but wanted to ask you:
naming them like so is just a convetion for the sessionStorage keys as shown using :
console.log(Object.keys(sessionStorage));
or they must be the same names of the VARIABLES receiving the value from the input?

**Pull request merged**